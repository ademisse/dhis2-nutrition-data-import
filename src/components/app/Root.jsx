import React from "react";
import PropTypes from "prop-types";
import i18n from "@dhis2/d2-i18n";
import "rc-tree-select/assets/index.css";
import OUMapList from "../oumap/OUMapList";
import SettingsIcon from "@material-ui/icons/Settings";
import SystemUpdateIcon from "@material-ui/icons/SystemUpdate";
import ImportDataTemplate from "../ImportData/ImportDataTemplate";
import FileUpload from "../ImportData/FileUpload/FileUpload";
import Tabs, { Tab } from "material-ui/Tabs";

import Paper from "@material-ui/core/Paper";
import { withFeedback, levels } from "../feedback";

class Root extends React.Component {
    static propTypes = {
        d2: PropTypes.object.isRequired,
        feedback: PropTypes.func.isRequired,
    };

    showFeedback = () => {
        this.props.feedback(levels.INFO, i18n.t("Hello there"));
    };

    render() {
        const { d2 } = this.props;

        return (
            <React.Fragment>
                {/* <p>Current user: {d2.currentUser.displayName}</p> */}

                <Tabs d2={d2}>
                    <Tab label={i18n.t("Data Templates")} icon={<SettingsIcon />}>
                        <Paper style={styles.paper}>
                            <ImportDataTemplate />
                        </Paper>
                    </Tab>
                    <Tab label={i18n.t("Organisation Unit Mapping")} icon={<SettingsIcon />}>
                        <Paper style={styles.paper}>
                            <OUMapList d2={d2} />
                        </Paper>
                    </Tab>
                    <Tab label={i18n.t("Import Data")} icon={<SystemUpdateIcon />}>
                        <Paper style={styles.paper}>
                            <FileUpload d2={d2} />
                        </Paper>
                    </Tab>
                </Tabs>
            </React.Fragment>
        );
    }
}
const styles = {
    paper: {
        padding: "0px",
        paddingTop: "10px",
    },
};
export default withFeedback(Root);
