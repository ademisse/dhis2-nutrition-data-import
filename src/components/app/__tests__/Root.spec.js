import React from "react";
import renderer from "react-test-renderer";

import { getD2Stub } from "../../../utils/testing";
import Root from "../Root";

it("renders correctly", () => {
    const props = { feedback: jest.fn(), d2: getD2Stub() };
    const tree = renderer.create(<Root {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
});
