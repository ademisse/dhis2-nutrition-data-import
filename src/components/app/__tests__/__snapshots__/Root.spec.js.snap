// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <p>
    Current user: 
    John Traore
  </p>,
  <p>
    Add your App components here
  </p>,
  <button
    onClick={[Function]}
  >
    Click to show feedback
  </button>,
]
`;
